futures-util/src/io/buf_reader.rs

```diff
@@ -71,6 +58,21 @@ impl<R: AsyncRead> BufReader<R> {
     }
 }
 
+impl<R: AsyncRead> BufReader<R> {
```

Comment:
I think adding the `} impl<R> BufReader<R> {` after the constructor definitions, rather than moving the constructor definitions, would make the diff smaller and make the document more readable. (I think the order constructor->accessor is usually more readable because it is the same as the actual order in the program.)